LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -1095,12 +1097,57 @@ namespace WPEFramework
                 return;
 
             if (data_ptr && owner_str
-                    && (strcmp(owner_str, IARM_BUS_CECHOST_NAME) == 0))
+                    && (strncmp(owner_str, IARM_BUS_CECHOST_NAME, strlen(IARM_BUS_CECHOST_NAME)) == 0))
             {
                 _instance->onDeviceStatusUpdateEnd(eventId, data_ptr, len);
             }
         }
 
+        void LgiHdmiCec::onCecStatusChange(IARM_EventId_t eventId, const void* data_ptr, size_t len)
+        {
+            assert(data_ptr != NULL);
+            if (len < sizeof(IARM_Bus_CECHost_DevMgrStatus_Param_t))
+            {
+                return;
+            }
+
+            const IARM_Bus_CECHost_DevMgrStatus_Param_t* eData = static_cast<const IARM_Bus_CECHost_DevMgrStatus_Param_t*>(data_ptr);
+
+            LOGINFO("status=%s-%s\n", eData->status ? "true" : "false", cecEnableStatus ? "true" : "false");
+            if (cecEnableStatus != eData->status)
+            {
+                std::lock_guard<std::mutex> guard(m_mutex);
+
+                cecEnableStatus = eData->status;
+                if (cecEnableStatus == false)
+                {
+                    m_devices.clear();
+                    m_scan_devices.clear();
+                    m_rescan_in_progress = false;
+                }
+                else
+                {
+                    m_rescan_in_progress = true;
+                    m_scan_id++;
+                    requestRescanning(m_scan_id);
+                }
+            }
+        }
+
+        void LgiHdmiCec::cecHostCecStatusChange(const char *owner_str, IARM_EventId_t eventId, void* data_ptr, size_t len)
+        {
+            LOGINFO("owner=%s", owner_str? owner_str: "<<UNKNOWN>>");
+
+            if (!LgiHdmiCec::_instance)
+                return;
+
+            if (data_ptr && owner_str
+                    && (strncmp(owner_str, IARM_BUS_CECHOST_NAME, strlen(IARM_BUS_CECHOST_NAME) + 1) == 0))
```

Comment:
why strlen(IARM_BUS_CECHOST_NAME) + 1 ? especially since it's just:
strncmp(owner_str, IARM_BUS_CECHOST_NAME, strlen(IARM_BUS_CECHOST_NAME)) 
on l. 1043 & 1100